src/MockBot.Api/Controllers/ChatController.cs

```diff
@@ -59,16 +59,25 @@ public async Task<ActionResult> PostMessageAsync(Guid chatId)
         {
             try
             {
-                string content;
-                using (StreamReader reader = new(Request.Body, Encoding.UTF8))
+                using StreamReader reader = new(Request.Body, Encoding.UTF8);
+                string content = await reader.ReadToEndAsync().ConfigureAwait(false);
+
+                if (string.IsNullOrEmpty(content))
                 {
-                    content = await reader.ReadToEndAsync().ConfigureAwait(false);
+                    return new BadRequestObjectResult("Post must have a body");
```

Comment:
At some point we should fix this to just use ordinary MVC deserialization.  Not in this PR - but this looks odd.

Reply:
Yes, totally agree. I think the whole project would benefit from a refactor, but I was trying to avoid doing too much of that for this PR and focus just on what was required to meet the objectives of the ticket. I've captured a low priority issue to track this and other refactoring work: https://github.com/buerokratt/Mock-Bot/issues/36